### ISSUE TITLE
Tuning yolo4_darknet on COCO

### ISSUE THREAD

**Issue by negvet** [david8862/keras-YOLOv3-model-set#149]
Hi @david8862

I am trying to tune yolo4_darknet on COCO and not able to reproduce the metrics. I do not see any example in the readme table with the darknet backbone. Did you try to train with darknet? 

Below are the steps to reproduce:

- Dump the init model

`python yolo.py --image --model_type=yolo4_darknet --weights_path=weights/yolov4.h5 --anchors_path=configs/yolo4_anchors.txt  --output_model_file=yolo-v4-init-dumped.h5 --model_image_size=608x608 --dump_model`

- Eval the dump model. The results:

![image](https://user-images.githubusercontent.com/17028475/107208671-fbb79480-6a12-11eb-84dc-627e1d1f46b8.png)

I am happy with these results. Want to get similar after several epochs of lightweight tuning.

- Train (to make things simpler and to keep memory under control I froze the backbone for the fine-tuning stage as well - change the code for that)

`python train.py --model_type=yolo4_darknet --anchors_path=configs/yolo4_anchors.txt --annotation_file=train2017.txt --val_annotation_file=val2017.txt --classes_path=configs/coco2017_origin_classes.txt --save_eval_checkpoint --gpu_num 3 --model_image_size=608x608 --learning_rate 0.000001 --decay_type=cosine --freeze_level=1 --weights_path=yolo4-init-dumped.h5 --transfer_epoch=0 --total_epoch=1`

- Dump of the trained model

`python yolo.py --model_type=yolo4_darknet --anchors_path=configs/yolo4_anchors.txt --classes_path=configs/coco2017_origin_classes.txt --model_image_size=608x608 --dump_model --weights_path=logs/000/trained_final.h5 --output_model_file=logs/000/dumped.h5`

- Eval the tuned model

`python eval.py --anchors_path=configs/yolo4_anchors.txt --classes_path=configs/coco2017_origin_classes.txt --model_image_size=608x608 --eval_type=COCO --iou_threshold=0.5 --conf_threshold=0.001 --annotation_file=val2017.txt --save_result --model_path=logs/000/dumped.h5`

I expected to get similar results (w.r.t. init untuned dump model) due to the small learning rate. But I got the following after 1 epoch:

![image](https://user-images.githubusercontent.com/17028475/107208874-33bed780-6a13-11eb-932a-d678ac64f322.png)

I also tried bigger LR and more epochs (several epochs for --transfer_epoch and 6-8 for --total_epoch) but was able to get only 40-43%.

**Comment by david8862** [comment]
@negvet I didn't try to tuning the official yolo4_darknet model, since it has been a well trained model and may not easy to get a better result. If you want to have a try, you may need to add related YOLOv4 tricks like "Eliminate grid sensitivity/Multi anchor assignment/Mosaic data augment", which is mentioned in YOLOv4 paper.

**Comment by negvet** [comment]
@david8862 thank you for the reply. I agree that reproducing the trained darknet results from scratch can be hard, which requires mosaic augmentation and other tricks. 

I want to pay your attention to the fact that a small learning rate (like 1e-6) should not introduce any significant changes to the model during tuning.  Therefore, the metric should keep more or less the same. But it drops instead for several %. Probably it is a bug.

Best

**Comment by negvet** [comment]
After adding YOLOv4 tricks as mentioned above "Eliminate grid sensitivity/Multi anchor assignment/Mosaic data augment" the model converged on the lr=1e-6 to the init metrics, even with batch size 16.
